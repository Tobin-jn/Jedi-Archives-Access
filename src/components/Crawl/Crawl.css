aside {
  margin: auto;
}

.header-logo-wrapper {
  width: 70px;
  margin: 2rem auto 0;
}

.header-logo {
  color: #ADD8E6;
  font-size: 80px;
}

.landing-loading h3 {
  font-family: Andale Mono,AndaleMono,monospace;
  color: #ADD8E6;
  text-align: center;
  letter-spacing: 5px;
  text-transform: uppercase;
}

.thumbprint {
  height: 65px;
  width: 50px;
  border-top-left-radius: 50px;
  border-top-right-radius: 50px;
  border-bottom-left-radius: 50px;
  border-bottom-right-radius: 50px;
  border: #ADD8E6 1px solid;
  margin: 2rem auto 0;
  transition: 1s;
}

.thumbprint img {
  padding-top: 5px;
  opacity: 0;
  height: 55px;
  width: 45px;
  transition: 1s;
  display: block;
  margin: auto;
}

.thumbprint:hover {
  cursor: pointer;
}

.thumbprint:hover img {
  opacity: 1;
}

.active-thumbprint img {
  opacity: 1;
}

.active-thumbprint {
  box-shadow: 0 3px 10px #ADD8E6;
}

.securing,
.handshake,
.welcome,
.briefing {
  text-align: center;
  margin: 0 auto;
  font-size: .8rem;
  position: relative;
  font-family: Andale Mono,AndaleMono,monospace;
  text-transform: uppercase;
  letter-spacing: 5px;
  color: transparent;
}



.securing {
  width: 274px;
} 

.handshake {
  width: 253px;
}

.welcome {
  width: 253px;
}

.briefing {
  margin: 2rem auto;
  width: 290px;
}

.securing:before,
.handshake:before,
.welcome:before,
.briefing:before {
  margin: 0;
  text-align: center;
  content: attr(data-text);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  color: transparent;
  overflow: hidden;
  white-space: nowrap;
}

.briefing:before {
  padding: .5rem  0;
}

.securing:before {
  width: 274px;
}

.print-securing:before {
  text-align: center;
  color: #ADD8E6;
  animation: type 3s steps(16);
}

.print-handshake:before {
  text-align: center;
  color: #ADD8E6;
  animation: type 3s steps(15);
}

.print-welcome:before {
  text-align: center;
  color: #ADD8E6;
  animation: type 3s steps(14);
}

.print-briefing:before {
  text-align: center;
  color: #ADD8E6;
  animation: type 3s steps(17); 
}

@keyframes type {
  0% {
    width: 0;
  }
  50% {
    width: 50%;
  }
  100% {
    width: 100%;
  }
}

.crawl-text-wrapper {
  opacity: 0;
  color: #ADD8E6;
  transition: .5s;
}

.show-crawl {
  opacity: 1;
}

.hide-welcome {
  display: none;
}

.crawl-text-wrapper h2 {
  text-align: center;
  text-transform: uppercase;
}

.smol-text {
  font-size: .5rem;
}