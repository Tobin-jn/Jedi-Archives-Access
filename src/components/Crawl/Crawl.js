import React, {Component} from 'react';
import './Crawl.css';
import PropTypes from 'prop-types';
import Error from '../Error/ErrorPage';
import Briefing from '../Briefing/Briefing';
import Loading from '../Loading/Loading'

class Crawl extends Component {
  constructor() {
    super()
    this.state = {
      thumbprint: '',
      securing: '',
      handshake: '',
      welcome: '',
      briefing: '',
      fadeWelcome: '',
      hideWelcome: '',
      showCrawl: '',
      ready: false,
      error: ''
    }
  }

  checkReady = () => {
    if (this.state.ready && !this.props.error) {

      this.setState({
        showCrawl: 'show-crawl',
        hideWelcome: 'hide-welcome'
      })

      if (window.innerWidth > 475 && this.props.hamburger.status === 'closed') {
        this.props.hamburgerChange();
      }

    } else if (this.state.ready && this.props.error) {
      this.setState({
        error: 'display-crawl-error',
        hideWelcome: 'hide-welcome'
      })
    }
  }

  handleFingerPrint = async () => {
    await this.setState({thumbprint: 'active-thumbprint'})
    await setTimeout(this.printConnect, 500);
    await setTimeout(this.printHandshake, 2500);
    await setTimeout(this.printWelcome, 5000);
    await setTimeout(this.printBriefing, 6500);
    await setTimeout(this.props.setReady, 6500);
  }

  printConnect = async () => {
    await this.setState({securing: 'print-securing'})
  }

  printHandshake = async () => {
    await this.setState({handshake: 'print-handshake'})
  }

  printWelcome = async () => {
    await this.setState({welcome: 'print-welcome'})
  }

  printBriefing = async () => {
    await this.setState({
      briefing: 'print-briefing',
      ready: true
    })
  }

  render() {
    return (
      <aside 
        onClick={() => {this.checkReady()}}
        aria-label='landing-screen'
      >
        <section className='header-section'>
          <div className='header-logo-wrapper'>
            <i className='fab fa-jedi-order header-logo'></i>
          </div>
        </section>
        <main className={`main-landing-section ${this.state.fadeWelcome} ${this.state.hideWelcome}`}>
          <section className='landing-loading'>
            <h3>Imprint thumb to connect</h3>
          </section>
          <h3 
            data-text='Securing connection...' 
            className={`securing ${this.state.securing}`}
          >
            Securing connection...
          </h3>
          <h3 
            className={`handshake ${this.state.handshake}`}
            data-text='Handshake successful!'
          >
            Handshake successful.
          </h3>
          <h3 
            className={`welcome ${this.state.welcome}`}
            data-text='Welcome, Master Jedi'
          >
            Welcome, Master Jedi.
          </h3>
          <div 
            className={`thumbprint ${this.state.thumbprint}`}
            onClick={() => {this.handleFingerPrint()}}
          >
            <img src={'./fingerprint.png'} alt='fingerprint' />
          </div>
          <h2 
            data-text='Tap to receive briefing'
            className={`briefing ${this.state.briefing}`}
          >
            Tap to receive briefing
          </h2>
        </main>
        <main 
          className={`crawl-text-wrapper ${this.state.showCrawl}`}
          aria-label='post-login-screen'
        >
          {!this.props.loading && <Briefing 
            crawl={this.props.film.opening_crawl}
            title={this.props.film.title}
            id={this.props.film.episode_id}
          />}
          {this.props.loading && <Loading />}

        </main>
        <Error errorState={this.state.error}/>
      </aside>
    );
  }
}
  
Crawl.propTypes = {
  film: PropTypes.object.isRequired,
  setReady: PropTypes.func.isRequired,
  error: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired
};

export default Crawl;