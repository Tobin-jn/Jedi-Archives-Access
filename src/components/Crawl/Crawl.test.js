import React from 'react';
import { shallow } from 'enzyme';
import Crawl from './Crawl';

describe('Crawl', () => {
  let wrapper;
  let defaultState = {
      thumbprint: '',
      securing: '',
      handshake: '',
      welcome: '',
      briefing: '',
      fadeWelcome: '',
      hideWelcome: '',
      showCrawl: '',
      ready: false,
      error: ''
  }

  const mockHamburger = {status: 'closed'}

  beforeEach(() => {

    let mockReady = jest.fn();
    let mockHamburgerChange = jest.fn()

    wrapper = shallow(<Crawl film={{}}
                        setReady={mockReady}
                        error={false}
                        loading={false} 
                        hamburger={mockHamburger}
                        hamburgerChange={mockHamburgerChange}
                      />);
                        
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have default state', () => {
    expect(wrapper.state()).toEqual(defaultState)
  })

  it('should call checkReady onclick', () => {
    const spy = spyOn(wrapper.instance(), 'checkReady')

    wrapper.find('aside').simulate('click');

    expect(spy).toHaveBeenCalled()
  });

  it('should setState if ready and there is no error', async () => {
    await wrapper.setState({
      ready: true,
    });

    wrapper.instance().checkReady();

    expect(wrapper.state().showCrawl).toEqual('show-crawl')
    expect(wrapper.state().hideWelcome).toEqual('hide-welcome')
  });

  it('should setState if ready and there is an error', async () => {
    wrapper = shallow(<Crawl film={{}} setReady={jest.fn()} error={true} loading={false}/>)

    await wrapper.setState({
      ready: true,
    });

    wrapper.instance().checkReady();

    expect(wrapper.state().error).toEqual('display-crawl-error')
    expect(wrapper.state().hideWelcome).toEqual('hide-welcome')
  });

  it('should call several functions when handleFingerPrint is called', async () => {
    const spyOne = spyOn(wrapper.instance(), 'printConnect')
    const spyTwo = spyOn(wrapper.instance(), 'printHandshake')
    const spyThree = spyOn(wrapper.instance(), 'printWelcome')
    const spyFour = spyOn(wrapper.instance(), 'printBriefing')
    const mockSetReady = jest.fn()

    wrapper = shallow(<Crawl film={{}} setReady={mockSetReady} error={true} loading={false}/>)

    await wrapper.instance().handleFingerPrint();

    expect(setTimeout(((spyOne).toHaveBeenCalled), 7000));
    expect(setTimeout(((spyTwo).toHaveBeenCalled), 7000));
    expect(setTimeout(((spyThree).toHaveBeenCalled), 7000));
    expect(setTimeout(((spyFour).toHaveBeenCalled), 7000));
    expect(setTimeout(((mockSetReady).toHaveBeenCalled), 7000));
  });

  it('should call setState when handleFingerPrint is called', async () => {
    await wrapper.instance().handleFingerPrint()

    expect(wrapper.state().thumbprint).toEqual('active-thumbprint');
  });

  it('should call setState on printConnect', async () => {
    await wrapper.instance().printConnect();

    expect(wrapper.state().securing).toEqual('print-securing');    
  });

  it('should call setState on printHandshake', async () => {
    await wrapper.instance().printHandshake();

    expect(wrapper.state().handshake).toEqual('print-handshake');    
  });

  it('should call setState on printWelcome', async () => {
    await wrapper.instance().printWelcome();

    expect(wrapper.state().welcome).toEqual('print-welcome');    
  });

  it('should call setState on printBriefing', async () => {
    wrapper.instance().printBriefing();

    expect(wrapper.state().briefing).toEqual('print-briefing');  
    expect(wrapper.state().ready).toEqual(true);    

  });

  it('should call handleFingerPrint when clicked', () => {
    const spy = spyOn(wrapper.instance(), 'handleFingerPrint')

    wrapper.find('.thumbprint').simulate('click');

    expect(spy).toHaveBeenCalled();
  })
})