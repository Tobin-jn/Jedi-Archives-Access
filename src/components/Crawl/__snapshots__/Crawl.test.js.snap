// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crawl should match snapshot 1`] = `
<aside
  aria-label="landing-screen"
  onClick={[Function]}
>
  <section
    className="header-section"
  >
    <div
      className="header-logo-wrapper"
    >
      <i
        className="fab fa-jedi-order header-logo"
      />
    </div>
  </section>
  <main
    className="main-landing-section  "
  >
    <section
      className="landing-loading"
    >
      <h3>
        Imprint thumb to connect
      </h3>
    </section>
    <h3
      className="securing "
      data-text="Securing connection..."
    >
      Securing connection...
    </h3>
    <h3
      className="handshake "
      data-text="Handshake successful!"
    >
      Handshake successful.
    </h3>
    <h3
      className="welcome "
      data-text="Welcome, Master Jedi"
    >
      Welcome, Master Jedi.
    </h3>
    <div
      className="thumbprint "
      onClick={[Function]}
    >
      <img
        alt="fingerprint"
        src="./fingerprint.png"
      />
    </div>
    <h2
      className="briefing "
      data-text="Tap to receive briefing"
    >
      Tap to receive briefing
    </h2>
  </main>
  <main
    aria-label="post-login-screen"
    className="crawl-text-wrapper "
  >
    <Briefing />
  </main>
  <ErrorPage
    errorState=""
  />
</aside>
`;
